examples/core/instancing/app.js

```diff
@@ -68,7 +70,8 @@ void main(void) {
   // Set up data for modules
   color = instanceColors;
   project_setNormal(normal);
-  picking_setPickingColor(vec3(0., instancePickingColors));
+  vec4 pickColor = vec4(0., instancePickingColors, 1.0);
+  LUMAGL_pickColor(pickColor);
```

Comment:
The instancing app is luma.gl's signature app, perhaps we should not use it as a playground for demoing new APIs? Maybe choose another app? Then the rather arbitrary picking hook would stand out less...

Reply:
I'd suggest the signature app should show off as much of the API as possible.

I'll take out the "LUMAGL" prefixes, because it gives the impression that it's official API, which isn't the case. As for whether a pick color hook should be defined, it's really up to the application, but in general, the idea is that the hooks should allow module functionality to be inserted and removed from a shader without having to modify the shader code at all. Perhaps there's a more generic hook that could be defined, but if the module is to be "drop in" it will need some way to get pick color data from a hook function.

Reply:
> I'd suggest the signature app should show off as much of the API as possible.

Well, yes, we do want demos for the new APIs.

My point was that adding shader injection code to this example seems like an unnecessary distraction/complication to a new user (or even an experienced user) trying to read it and understand how to duplicate the behavior after seeing it on our home page.

And add 3-4 more features like this in that demo and the source code could become rather incomprehensible to a new user.

Shouldn't most demos should be designed to be as simple as possible, to do a specific task in as few lines as code as possible, and "show off" our API in that way (demonstrating its expressiveness / compactness / simplicity etc)?

That said, we should perhaps have some app(s) that attempt to show how to use all our different APIs. I would personally prefer that it was not the `instancing` app that is the main demo on our home page. In this case, some more "shader heavy" example focusing on power use of our "shadertools" module might be appropriate.

Reply:
I think there's a longer discussion to be had about the purpose of the examples. They're currently not very good at presenting any interesting parts of the API. For 8.0 I'd like to rethink the entire set to focus on small API examples, plus some showcases that put the pieces together into something interesting (with instancing, to me, fitting in the latter category).